drivers/spi/Kconfig.xec_qmspi_v2

```diff
@@ -0,0 +1,18 @@
+# Microchip XEC QMSPI FULL DUPLEX, QMSPI LDMA
+
+# Copyright (c) 2022 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+choice
```

Comment:
This seems to be 2 distrnict drivers so `choice` doesn't make sense.

Should just be something like:

```
config SPI_XEC_QMSPI_LDMA
   prompt "QMSPI driver with LDMA (local dma)"
   default y
   depends on DT_HAS_MICROCHIP_XEC_QMSPI_LDMA_ENABLED
   ... 

config SPI_XEC_QMSPI_FULL_DUPLEX
   prompt "Use QMSPI full duplex driver to work with Zephyr NOR driver"
   default y
   depends on DT_HAS_MICROCHIP_XEC_QMSPI_FULL_DUPLEX_ENABLED
   ...

Reply:
choice is required because only one driver can work at a time, both work with the same hardware

see discussion here https://github.com/zephyrproject-rtos/zephyr/pull/46273#discussion_r896078492

Reply:
this driver was added to support full duplex transfers as expected by the Zephyr spi driver model to work with Zephyr SPI NOR driver. This cannot be added with existing driver as a feature.

Reply:
> This cannot be added with existing driver as a feature.

Why?

Seems like there is at least common defines, structs, etc that should NOT be duplicated.

Reply:
This full duplex driver is compatible with SPI NOR driver which expects on every spi clock we transmit one bit and receive one bit. Here for every transmit buffer you need a receive buffer.

The ldma driver supports quad and dual mode transfers using local dma and uses a model where it processes all transmit buffers and then process all receive buffers, if any.

Merging is complex.

This full duplex driver is required for our Zephyr chromium integration which is on hold due to this PR since June.

Reply:
@galak we also consider the same request that you propose but due the low level operations between the 2 drivers, the result will be a merged one plagued with ifdefs everywhere which would be hard to maintain and debug.

Reply:
> @galak we also consider the same request that you propose but due the low level operations between the 2 drivers, the result will be a merged one plagued with ifdefs everywhere which would be hard to maintain and debug.

I don't understand this comment.  If its the same hardware, than the structs, defines, etc that are being used would be the same.

Reply:
we had discussion with @MaureenHelm and agreed to keep the 2 different drivers but update the full_duplex driver to avoid the duplicate `qspi_regs` sturcture

Reply:
updated to use the common qmspi_regs structure

Reply:
also cleaned up the Kconfig